Tasks/DownloadPackageV1/package.ts

```diff
@@ -144,20 +144,23 @@ export abstract class Package {
     ): Promise<string> {
         const routeValues = {
             feedId: feedId,
-            project: project
+            project: project,
+            packageId: packageId
         };
         const queryParams = {
-            packageIdQuery: packageId,
             protocolType: this.packageProtocolAreaName
         };
 
         return new Promise<string>(async (resolve, reject) => {
             this.getPackageMetadata(this.feedConnection, routeValues, queryParams, this.getPackagesAreaId)
             .then(packages => {
-                tl.debug("Found " + packages["count"] + " packages matching search pattern " + packageId);
-                for (let i = 0; i < packages["count"]; i++) {
-                    if (packages["value"][i]["id"] == packageId && packages["value"][i]["versions"][0]["isListed"]) {
-                        return resolve(packages["value"][i]["versions"][0]["normalizedVersion"]);
+                if(packages["id"] == packageId){
+                    var versions = packages["versions"];
```

Comment:
use `let` or `const` instead of `var`. `var` is an obsolete way to define variables in JS.